EventGenerator/src/Pileup_module.cc

```diff
@@ -102,7 +114,18 @@ namespace mu2e {
          <<", decay fraction = "<<decayFraction_
          <<std::endl;
     }
-
+    if(makeHistograms_){
```

Comment:
I don't see why these histograms/tuples need to be made inside the generator; i think you could move these to an analyzer that reads the primary and produce the same plots.  That provides separation of function, which is a sign of good design.  Even better, you could use the same analyzer for any primary generator, so you avoid code duplication.

Reply:
Ok, I can look into that. I think the old generators used to have something like this (at least for RPC they did). I'll look into another function

Reply:
You are right that we used to do it that way.   We decided to move away from that to the model that Dave described.   Ideal one analyzer should be able to make histograms for many generators.

Reply:
This comment applies to both this module and your new module.